File: 백준/Silver/11726. 2×n 타일링/2×n 타일링.py
def fact(num):
  total=1
  for i in range(1,num+1):
    total*=i
  return total
    
n=int(input())
end=n//2
result=0
for i in range(end+1):
  result+=(fact(n-i)//(fact(n-i*2)*fact(i)))%10007

print(result%10007)

File: 백준/Silver/11726. 2×n 타일링/2×n 타일링.js
const fs = require("fs");
let n = BigInt(fs.readFileSync(0).toString().trim()); // 입력을 BigInt로 변환
let end = n / 2n; // BigInt 연산

const fact = (num) => {
    let total = BigInt(1);
    for (let i = 1n; i <= num; i++) {
        total *= i; // BigInt 곱셈
    }
    return total;
};

let result = 0n; // 결과값도 BigInt로 초기화
for (let i = 0n; i <= end; i++) {
    // 팩토리얼 계산 및 조합식 연산
    result += (fact(n - i) / (fact(n - i * 2n) * fact(i))) % 10007n;
}
console.log(Number(result % 10007n)); // 결과를 Number로 변환하여 출력
